interpreter/planner.go

```diff
@@ -555,15 +552,7 @@ func (p *planner) planCreateStruct(expr *exprpb.Expr) (Interpretable, error) {
 // planCreateObj generates an object construction Interpretable.
 func (p *planner) planCreateObj(expr *exprpb.Expr) (Interpretable, error) {
 	obj := expr.GetStructExpr()
-	typeName := obj.MessageName
-	var defined bool
-	for _, qualifiedTypeName := range p.pkg.ResolveCandidateNames(typeName) {
-		if _, found := p.provider.FindType(qualifiedTypeName); found {
-			typeName = qualifiedTypeName
-			defined = true
-			break
-		}
-	}
+	typeName, defined := p.resolveTypeName(obj.MessageName)
```

Comment:
So this all looks okay, but it's largely duplicating the logic in the checker.  I would much rather see the planner and the checker both call a separate pass of walking the AST to do the qualified identifier rewrites, with a pluggable qualified name tester that checks either declarations (checker) or bound variables and dispatchable functions (planner).  Yet another function can rewrite a CheckedExpr based on its reference map, returning early if it sees that the rewrite has already been performed.

Oh, nasty thought - can the iteration variable in a comprehension (fold) be a qualified name?  What if we are in container "a.b" and write `mylist.exists(a.b.c, c % 2 == 0)`?